pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -38,6 +43,59 @@ func (e *contentRemainingError) Error() string {
 	return fmt.Sprintf("some content remains in the namespace, estimate %d seconds before it is removed", e.Estimate)
 }
 
+// operation is used for caching if an operation is supported on a dynamic client.
+type operation string
+
+const (
+	operationDeleteCollection operation = "deleteCollection"
+	operationList             operation = "list"
+)
+
+// operationSupportedCacheEntry is an entry in the cache.
+type operationSupportedCacheEntry struct {
+	op                   operation
+	groupVersionResource unversioned.GroupVersionResource
+	supported            bool
+}
+
+// operationSupportedCache is a simple cache to remember if an operation is supported for a resource.
+type operationSupportedCache struct {
+	store cache.Store
+}
```

Comment:
Maybe I'm missing something, but why do we need a full blown cache here? This doesn't seem to need any of the functionality there. Could this be replaced with:

```go
type opKey struct {
	op  operation
	gvr unversioned.GroupVersionResource
}

type operationSupportedCache map[opKey]bool

...
supported, exists := c[opKey{op: op, gvr: gvr}]
```

?